server/app/report.go

```diff
@@ -0,0 +1,228 @@
+	if err != nil {
+		return "", errors.Wrapf(err, "generateSurveyReport: failed to get survey by ID, surveyID: %s", surveyID)
+	}
+
+	key := utils.NewID()
+
+	rawResponseCSVFilePath, err := a.generateRawResponseCSV(survey, key)
+	if err != nil {
+		return "", err
+	}
+
+	surveyMetadataFilePath, err := a.generateSurveyMetadataFile(survey, key)
+	if err != nil {
+		return "", err
+	}
+
+	zipPath := path.Join(os.TempDir(), "survey_report", key, "survey_report.zip")
+	files := []string{rawResponseCSVFilePath, surveyMetadataFilePath}
+	err = utils.CreateZip(zipPath, files)
+	if err != nil {
+		a.api.LogError("generateSurveyReport: failed to generate report zip file", "surveyID", surveyID, "error", err.Error())
+		return "", errors.Wrapf(err, "generateSurveyReport: failed to generate report zip file, surveyID: %s", surveyID)
+	}
+
+	// delete the unneeded files now
+	for _, filePath := range files {
+		a.api.LogDebug("generateSurveyReport: deleting file", "filePath", filePath)
+		err = os.Remove(filePath)
+		if err != nil {
+			// not a critical error, no need to break
+			a.api.LogWarn("generateSurveyReport: failed to delete file path", "filePath", filePath, "error", err.Error())
+		}
+	}
+
+	return zipPath, nil
+}
+
+func (a *UserSurveyApp) generateRawResponseCSV(survey *model.Survey, key string) (string, error) {
+	var lastResponseID string
+
+	headers := []string{"User ID", "Submitted At"}
+	for _, question := range survey.SurveyQuestions.Questions {
+		headers = append(headers, question.Text)
+	}
+
+	part := 0
+	for {
+		// get a page worth of results
+		data, err := a.store.GetAllResponses(survey.ID, lastResponseID, rawResponsePerPage)
+		if err != nil {
+			return "", err
+		}
+
+		if len(data) == 0 {
+			break
+		}
+
+		lastResponseID = data[len(data)-1].ID
+
+		// save them in a temp CSV
+		if err := a.saveTempCSVData(key, part, data, survey); err != nil {
+			return "", errors.Wrapf(err, "generateRawResponseCSV: surveyID: %s", survey.ID)
+		}
+
+		part++
+
+		if len(data) < rawResponsePerPage {
+			break
+		}
+	}
+
+	mergedFilePath, err := a.mergeParts(key, headers, part)
+	if err != nil {
+		return "", errors.Wrapf(err, "generateRawResponseCSV: failed to merge report parts, surveyID: %s", survey.ID)
+	}
+
+	// now that the parts have been merged, we can delete the
+	// dir where we were storing the parts
+	partDir := path.Join(os.TempDir(), "survey_report", key, "parts")
+	a.api.LogDebug("generateRawResponseCSV: deleting the part dir", "partDir", partDir)
+	if err := os.RemoveAll(partDir); err != nil {
+		a.api.LogError("generateRawResponseCSV: failed to delete part dir", "partDir", partDir, "error", err.Error())
+		// no need to break here, this is not a critical error
+	}
+
+	return mergedFilePath, nil
+}
+
+func (a *UserSurveyApp) saveTempCSVData(key string, part int, surveyResponses []*model.SurveyResponse, survey *model.Survey) error {
+	var buf bytes.Buffer
+
+	// TODO set important CSV attributes here
```

Comment:
What would these be?

Reply:
The defaults were fine, no need of setting any custom attribute. I've removed the comment now.